prow/spyglass/lenses/buildlog/buildlog.ts

```diff
@@ -46,16 +46,45 @@ function ansiToHTML(orig: string): string {
   });
 }
 
-async function replaceElementWithContent(element: HTMLDivElement) {
+interface ArtifactRequest {
+  artifact: string;
+  bottom: number;
+  length: number;
+  offset: number;
+  startLine: number;
+  top: number;
+}
+
+async function replaceElementWithContent(element: HTMLDivElement, top: number, bottom: number) {
+  // +length => make it a number (as opposed to a string)
+  // length! => I know this value is non-null
   const {artifact, offset, length, startLine} = element.dataset;
-  const content = await spyglass.request(JSON.stringify({
-    artifact, length: +length!, offset: +offset!, startLine: +startLine!}));
-  element.innerHTML = ansiToHTML(content);
-  fixLinks(element);
+  const r: ArtifactRequest = {
+    artifact: artifact!,
+    bottom,
+    length: +length!,
+    offset: +offset!,
+    startLine: +startLine!,
+    top,
+  };
+  const content = await spyglass.request(JSON.stringify(r));
   showElem(element);
+  element.outerHTML = ansiToHTML(content);
+  fixLinks(document.documentElement);
+  for (const button of Array.from(document.querySelectorAll<HTMLDivElement>(".show-skipped"))) {
+    if (button.classList.contains("showable")) {
+      continue;
+    }
+    button.addEventListener('click', handleShowSkipped);
+    button.className = button.className + " showable";
```

Comment:
I think you can do element.classList.add('class') instead?

Reply:
Done